moveit_py/src/moveit/moveit_py_utils/include/moveit_py/moveit_py_utils/rclpy_time_typecaster.hpp

```diff
@@ -0,0 +1,89 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2022, Shobin Vinod
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Shobin Vinod */
+
+#include <pybind11/pybind11.h>
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp/serialization.hpp>
+#include <rclcpp/duration.hpp>
+
+namespace py = pybind11;
+
+namespace pybind11
+{
+namespace detail
+{
+template <>
+struct type_caster<rclcpp::Time>
+{
+  PYBIND11_TYPE_CASTER(rclcpp::Time, _("rclcpp::Time"));
+
+  // convert from rclpy::Time to rclcpp::Time
+  bool load(py::handle src, bool)
+  {
+    if (src.is_none())
+      return false;
+
+    // check to validate if the object is a rclcpp::Time object
+    if (!py::hasattr(src, "nanaoseconds") || !py::hasattr(src, "clock_type"))
```

Comment:
Did you tested this? With this typo, it shouldn't have worked!
```suggestion
    if (!py::hasattr(src, "nanoseconds") || !py::hasattr(src, "clock_type"))
```

Reply:
I tested the code before rebasing to the main branch. To do that, I cherry-picked the commits, did a hard reset of the branch, and force-pushed it. It looks like the typo was missed during the rebase process. Thanks for catching it and making the update!